### PR TITLE
Github actions

### PR DESCRIPTION
**Pull request by johnnynunez**
Added python versions
Added linter
Added precommit 
Added artifacts

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          python -m build --sdist ./PythonAPI
+          python -m pip install --progress-bar off ./PythonAPI/dist/*.tar.gz
+      - name: Run Simple Test Cases
+        run: |
+          python tests/test_cases.py
+      - name: Upload wheel
+        uses: actions/upload-artifact@v3
+        with:
+          name: pycocotools-${{ matrix.python-version }}-${{matrix.os}}-wheel
+          path: ./PythonAPI/dist/*.tar.gz
```
Comment (SpecLad) [inline .github/workflows/workflow.yml:71]: This step is titled "Upload wheel", but it actually uploads the sdist, not the wheel.

**Comment by johnnynunez** [comment]
image
done!

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install From Source
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install ./PythonAPI
```
Comment (ppwwyyxx) [inline .github/workflows/workflow.yml:21]: Please follow the existing CI and add `--use-feature=in-tree-build`. Same below.

**Review by ppwwyyxx — CHANGES_REQUESTED** [review]
Please only keep `.github/workflows/workflow.yml` and revert other changes. This project aims to keep in sync with the official cocoapi as much as possible. And because of that we will not fix any linter warnings.

File: .github/workflows/workflow.yml (hunk 2 above)
Comment (SpecLad) [reply .github/workflows/workflow.yml:21]: FWIW, `in-tree-build` doesn't exist anymore. It was made the default in pip 21.3 and removed in 22.1.

Comment (ppwwyyxx) [reply .github/workflows/workflow.yml:21]: Ah, OK. Thanks for sharing that!

**Comment by ppwwyyxx** [comment]
github actions were added now

**Pull request closed without merging**